includes/Modules/Analytics/Advanced_Tracking/Measurement_Events/CF7_Event_List.php

```diff
@@ -18,24 +18,46 @@
  * @ignore
  */
 final class CF7_Event_List extends Measurement_Event_List {
-	// TODO: Implement metadata callbacks.
 
 	/**
-	 * CF7_Event_List constructor.
+	 * Registers functionality through WordPress hooks.
 	 *
 	 * @since n.e.x.t.
 	 */
-	public function __construct() {
-		$event = new Measurement_Event(
+	public function register() {
+		add_filter(
+			'do_shortcode_tag',
+			function( $output, $tag, $attr ) {
+				if ( 'contact-form-7' == $tag ) {
+					$this->collect_cf7_shortcode( $attr['id'] );
```

Comment:
What happens if the shortcode is placed without specifying an `id`? I know this would be incorrect usage, but let's make sure this doesn't trigger a PHP notice. We need to be coding defensively, WP is a distributed environment where people do all sorts of weird things.

Reply:
This issue has been resolved.